File: EVR-FE/src/services/feedback.service.js
import apiClient from './api/axios';
import { ENDPOINTS } from './api/endpoints';

export const feedbackService = {
  createFeedback: async (feedbackData, userId) => {
    return apiClient.post(ENDPOINTS.CREATE_FEEDBACK, feedbackData, {
      params: { userId },
    });
  },

  getUserFeedback: async (userId) => {
    return apiClient.get(ENDPOINTS.GET_USER_FEEDBACK, {
      params: { userId },
    });
  },

  getContractFeedback: async (contractId) => {
    return apiClient.get(ENDPOINTS.GET_CONTRACT_FEEDBACK(contractId));
  },
};

File: EVR/src/main/java/com/group7/evr/service/FeedbackService.java
package com.group7.evr.service;

import com.group7.evr.entity.Feedback;
import com.group7.evr.entity.User;

import java.util.List;

public interface FeedbackService {

    public Feedback createFeedback(Feedback feedback, User user);

    public List<Feedback> getUserFeedback(Integer userId);

    public List<Feedback> getContractFeedback(Integer contractId);
}